src/common/droid-util.c

```diff
@@ -1170,29 +1045,108 @@ static void add_all_profiles(pa_droid_profile_set *ps, const pa_droid_config_hw_
+                                const pa_droid_config_output *low_latency_output,
+                                const pa_droid_config_output *media_latency_output,
+                                const pa_droid_config_input *builtin_input,
+                                const pa_droid_config_input *external_input) {
+
+    pa_droid_profile *p;
+
+    pa_assert(ps);
+    pa_assert(module);
+
+    pa_log_debug("New default profile");
+
+    p = profile_new(ps, module, "default", "Default profile");
+
+    if (primary_output)
+        pa_idxset_put(p->output_mappings, pa_droid_mapping_get(ps, PA_DIRECTION_OUTPUT, primary_output), NULL);
+    if (low_latency_output && primary_output != low_latency_output)
+        pa_idxset_put(p->output_mappings, pa_droid_mapping_get(ps, PA_DIRECTION_OUTPUT, low_latency_output), NULL);
+    if (media_latency_output && primary_output != media_latency_output && low_latency_output != media_latency_output)
+        pa_idxset_put(p->output_mappings, pa_droid_mapping_get(ps, PA_DIRECTION_OUTPUT, media_latency_output), NULL);
+    if (builtin_input)
+        pa_idxset_put(p->input_mappings, pa_droid_mapping_get(ps, PA_DIRECTION_INPUT, builtin_input), NULL);
+    if (external_input && builtin_input != external_input)
+        pa_idxset_put(p->input_mappings, pa_droid_mapping_get(ps, PA_DIRECTION_INPUT, external_input), NULL);
+
+    p->priority += DEFAULT_PRIORITY * (pa_idxset_size(p->output_mappings) + pa_idxset_size(p->input_mappings));
+    p->priority += primary_output ? DEFAULT_PRIORITY : 0;
+    pa_hashmap_put(ps->profiles, p->name, p);
+}
+
+static void auto_add_profiles(pa_droid_profile_set *ps, const pa_droid_config_hw_module *module) {
+    const pa_droid_config_output *output;
+    const pa_droid_config_input *input;
+
+    const pa_droid_config_output *primary_output        = NULL;
+    const pa_droid_config_output *low_latency_output    = NULL;
+    const pa_droid_config_output *media_latency_output  = NULL;
+
+    const pa_droid_config_input *builtin_input          = NULL;
+    const pa_droid_config_input *external_input         = NULL;
+
+    uint32_t input_devices;
+
+    pa_assert(ps);
+    pa_assert(module);
+
+    /* first find different output types */
+    SLLIST_FOREACH(output, module->outputs) {
+        if (output->flags & AUDIO_OUTPUT_FLAG_PRIMARY)
+            primary_output = output;
+
+#if defined(HAVE_ENUM_AUDIO_OUTPUT_FLAG_RAW)
+        else if (output->flags & AUDIO_OUTPUT_FLAG_RAW)
+            pa_log_debug("Ignore output %s with flag AUDIO_OUTPUT_FLAG_RAW", output->name);
+#endif
+
+        else if (output->flags & AUDIO_OUTPUT_FLAG_FAST)
+            low_latency_output = output;
+
+        else if (output->flags & AUDIO_OUTPUT_FLAG_DEEP_BUFFER)
+            media_latency_output = output;
+    }
+
+    SLLIST_FOREACH(input, module->inputs) {
+        input_devices = input->devices;
+#if AUDIO_API_VERSION_MAJ >= 2
+        input_devices &= ~AUDIO_DEVICE_BIT_IN;
```

Comment:
I can't figure out how/if dropping that bit from local variable changes anything...

Reply:
all AUDIO_DEVICE_IN_foo have that AUDIO_DEVICE_BIT_IN, so that is dropped from the local variable so that we get proper match when checking the devices later.